File: ios/Classes/scan/ScanViewFactory.m

#import "ScanViewFactory.h"
#import "ScanView.h"

static NSString * scanView=@"scanView";

@implementation ScanViewFactory{
 
    NSObject<FlutterBinaryMessenger>* _messenger;
}

- (instancetype)initWithMessenger:(NSObject<FlutterBinaryMessenger> *)messenger{
    self = [super init];
    if (self) {
        _messenger=messenger;
    }
    return self;
}

- (NSObject<FlutterMessageCodec> *)createArgsCodec{
    return [FlutterStandardMessageCodec sharedInstance];
}
- (NSObject<FlutterPlatformView> *)createWithFrame:(CGRect)frame viewIdentifier:(int64_t)viewId arguments:(id)args{
    ScanPlatformView * scanView=[[ScanPlatformView alloc]initWithFrame:frame viewindentifier:viewId arguments:args binaryMessenger:_messenger];
    return scanView;
    
}
@end

@interface ScanPlatformView()
@property(nonatomic , strong)ScanView * view;


@end
@implementation ScanPlatformView{

}

- (instancetype)initWithFrame:(CGRect)frame viewindentifier:(int64_t)viewId arguments:(id)args binaryMessenger:(NSObject<FlutterBinaryMessenger> *)messenger{
    if(self = [super init]){
        _view=[[ScanView alloc]initWithFrame:frame viewIdentifier:viewId arguments:args binaryMessenger:messenger];
        _view.backgroundColor=[UIColor clearColor];
        _view.frame=frame;
        
    }
    return self;
}
- (nonnull UIView *)view {
    return _view;
}


@end

File: ios/Classes/scan/ScanViewFactory.swift
import AVFoundation
import Flutter

class ScanViewFactory: NSObject<FlutterPlatformViewFactory> {
    private var scanView = "scanView"
    
    private weak var messenger: (NSObjectProtocol & FlutterBinaryMessenger)?
    
    
    override init(messenger: (NSObjectProtocol & FlutterBinaryMessenger)?) {
        super.init()
        self.messenger = messenger
    }
    
    
    override func createArgsCodec() -> (NSObjectProtocol & FlutterMessageCodec)? {
        return FlutterStandardMessageCodec.sharedInstance()
    }
    
    override func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> (NSObjectProtocol & FlutterPlatformView)? {
        let scanView = ScanPlatformView(frame: frame, viewindentifier: viewId, arguments: args, binaryMessenger: messenger)
        return scanView
        
    }
}

class ScanPlatformView: NSObject<FlutterPlatformView> {

    override init(withMessenger frame: CGRect, viewindentifier viewId: Int64, arguments args: Any?, binaryMessenger messenger: (NSObjectProtocol & FlutterBinaryMessenger)) {
        super.init()
        let view = ScanView(frame: frame, viewIdentifier: viewId, arguments: args, binaryMessenger: messenger)
        view.backgroundColor = UIColor.clear
        view.frame = frame
    }
}
